src/main/webapp/dev/js/common/instructor.js

```diff
@@ -358,14 +358,19 @@ function bindCourseDeleteAllLinks() {
         event.preventDefault();
 
         const $clickedLink = $(event.currentTarget);
-        const messageText = 'Are you sure you want to permanently delete all the courses? '
-                + 'This operation will delete all students and sessions in these courses. '
-                + 'All instructors of these courses will not be able to access them hereafter as well.';
+        const $softDeletedCourses = $('td[id^="recoverycourseid"]');
+        let messageText = `Are you sure you want to permanently delete all ${$softDeletedCourses.length} courses in `
+                + 'the <b>Recycle Bin</b>? This operation will delete all students and sessions in these courses. '
+                + 'All instructors of these courses will not be able to access them hereafter as well.<br><br>'
+                + 'ID of courses to be permanently deleted:<br>';
+        for (let i = 0; i < $softDeletedCourses.length; i += 1) {
+            messageText += `- ${$softDeletedCourses[i].innerText}<br>`;
```

Comment:
Use JQuery's `text()` if you are going with the approach of coupling with the DOM. Note that `data-*` attributes are a convenient way to pass data from the backend though.

Is there any reason why we are not using lists here as well?

Reply:
If I use `text()` then the original value in `messageText` will be erased.

Emm...for the rest, I'm not quite sure what you mean. Can give me an example of how to get the values in this line?

Reply:
> If I use text() then the original value in messageText will be erased.

Not quite sure what you meant by this.

> Emm...for the rest, I'm not quite sure what you mean. Can give me an example of how to get the values in this line?

I meant data attributes such as (this is a poor example, not advocating it): `<td id="recoverycoursename${i.index}" data-course-name="${recoveryCourse.courseName}">${recoveryCourse.courseName}</td>`

> Is there any reason why we are not using lists here as well?

As in unordered lists:
```html
<ul>
   <li>...</li>
   <li>...</li>
</ul>
```

Reply:
Emm...Lets keep the original design as unordered list is a bit ugly :-p

![9038-image-3](https://user-images.githubusercontent.com/22449685/44010852-8d54482e-9ee7-11e8-9fa2-7665f1a3c22f.png)

Reply:
I think this is fine but I don't have any strong opinions for either. @damithc which do you prefer?

However, even if you keep the original design you should be styling unordered lists instead of using strings.

Reply:
The extra comma is caused by a bug in the code?
a proper list is preferred but ok with string if the former is too much trouble.

Reply:
Done :)

![9038-image-4](https://user-images.githubusercontent.com/22449685/44013822-984e23fe-9efa-11e8-9a93-4b5963780b24.png)

Reply:
Nice. Some rephrasing:

>Are you sure you want to permanently delete the following 3 courses in the ...
>...
>~~ID of courses to be permanently deleted:~~
>* ....
>* ...